### PR TITLE
early break for some visible nodes

### PR DESCRIPTION
**Pull request by OsmondJiang**
refine the logic and fix typo

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5901)

### CODE REVIEW

File: src/docfx/lib/markdown/MarkdigUtility.cs [hunk 1]
```diff
@@ -158,32 +158,39 @@ public static string ToPlainText(this MarkdownObject containerBlock)
...
+            var visible = false;
             Visit(markdownObject, node =>
             {
                 switch (node)
                 {
                     case HtmlBlock htmlBlock:
                         foreach (var line in htmlBlock.Lines.Lines)
                         {
-                            visiable = visiable || VisibleHtml(line.Slice.Text);
+                            visible = visible || VisibleHtml(line.Slice.Text);
```
Comment (live1206) [inline src/docfx/lib/markdown/MarkdigUtility.cs:169]: would `visible |= VisibleHtml(line.Slice.Text);` just work?

**Comment by mingwli** [comment]
[build](https://dev.azure.com/ceapex/Engineering/_build/results?buildId=146961&view=results) already finished

File: src/docfx/lib/markdown/MarkdigUtility.cs [hunk 2]
```diff
@@ -158,32 +158,39 @@ public static string ToPlainText(this MarkdownObject containerBlock)
...
                         }
-                        return true;
+                        break;
                     case HtmlInline htmlInline:
-                        visiable = visiable || VisibleHtml(htmlInline.Tag);
-                        return false;
+                        visible = visible || VisibleHtml(htmlInline.Tag);
+                        break;
+                    case HeadingBlock headingBlock when headingBlock.Inline is null || !headingBlock.Inline.Any():
+                        visible = visible || false;
```
Comment (live1206) [inline src/docfx/lib/markdown/MarkdigUtility.cs:180]: no affection for `visible`?

Comment (OsmondJiang) [reply src/docfx/lib/markdown/MarkdigUtility.cs:180]: no, but has affections to the workflow

File: src/docfx/lib/markdown/MarkdigUtility.cs (hunk 1 above)
Comment (OsmondJiang) [reply src/docfx/lib/markdown/MarkdigUtility.cs:169]: both way should be OK?

File: src/docfx/lib/markdown/MarkdigUtility.cs (hunk 2 above)
Comment (live1206) [reply src/docfx/lib/markdown/MarkdigUtility.cs:180]: so you just need the `break` here?

Comment (OsmondJiang) [reply src/docfx/lib/markdown/MarkdigUtility.cs]: add comment instead

**Pull request merged**